.github/workflows/test.yml

```diff
@@ -9,6 +9,20 @@ on:
       - '.gitignore'
 
 jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          pip install -r otterai/requirements.txt
+
+      - name: Run tests
+        run: |
+          python -m unittest discover -s otterai/tests -p 'test_*.py'
```

Comment:
✅ Great use of unittest for running tests! Ensure that all critical paths are covered in your test cases.